README.md

```diff
@@ -173,40 +173,96 @@ Create the PR, using the upstream (parent) branch from `.git/machete` as the bas
 
 <br/>
 
+
+## FAQ
+
+#### I've run `git machete discover`... but the branch layout I see in `.git/machete` doesn't exactly match what I expected. Am I doing something wrong?
+
+[//]: # (For how to find Medium header anchors, see https://www.freecodecamp.org/news/how-to-link-to-a-specific-paragraph-in-your-medium-article-2018-table-of-contents-method-e66595fea549/)
+No! It's all right, `discover` is based on an (imperfect)
+[heuristic](https://medium.com/virtuslab/git-machete-strikes-again-traverse-the-git-rebase-jungle-even-faster-with-v2-0-f43ebaf8abb0#0544)
+which usually yields branch layout close to what the user expected...
+but it still might differ in details (like, branches discovered to be children of `main`/`develop` instead of each other).
+
+Just run [`git machete edit`](https://git-machete.readthedocs.io/en/stable/#edit) to fix the layout manually.
+If you're working on JetBrains IDEs, you can use [git-machete IntelliJ plugin](https://github.com/VirtusLab/git-machete-intellij-plugin#git-machete-intellij-plugin)
+for features like branch name completion when editing `.git/machete` file.
+
+Also, consider [`git machete github checkout-prs`](#github-integration) instead of `git machete discover` if you already have GitHub PRs opened.
+
+<br/>
+
+#### Can I use `git merge` for dealing with stacked PRs?
+
+Generally, there's a [trilemma](https://en.wikipedia.org/wiki/Trilemma) (_choose at most two out of three_):
+1. stacked PRs
+2. rebase/squash (generally: rewriting git history)
+3. merge commits (i.e. commits [with 2+ parents](https://slides.com/plipski/git-internals#/7))
+
+While git-machete supports merging parent branch (like `main`) to update a branch
+([`git machete traverse --merge`](https://git-machete.readthedocs.io/en/stable/#traverse)),
+it works poorly with stacked PRs.
+You might end up with a very tangled history very quickly, and a non-trivial sequence of `git cherry-pick`s might be needed to restore order.
+
+We recommend choosing only (1.) and (2.) from the trilemma, i.e. stacking PRs and using squash/rebase,
+and **not** using `git merge` (other than for [backporting hotfixes](https://slides.com/plipski/git-machete/#/11)).
+
+<br/>
+
+#### Sometimes when I run `update` or `traverse`, too many commits are taken into the rebase... how to fix that?
+
+Contrary to the popular misconception, git doesn't have a notion of ["commits belonging to a branch"](https://slides.com/plipski/git-internals#/41).
+A branch is just a movable pointer to a commit.
+This makes it hard in general case to determine the range of commits that form the "unique history" of the given branch.
+There's an entire algorithm in git-machete for determining the
+[_fork point_](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02#1ac9)
+of the branch (i.e. the place after which the unique history of the branch starts).
```

Comment:
Too many details imho. Simpler & shorter would imho work better:

```suggestion
This makes it hard to figure-out in all of the cases.
```

Reply:
Huh do you think this can stay as is, assuming the _edu_ variant?

Reply:
@PawelLipski Yes, if it's the edu variant it can stay

Reply:
Okay, leaving roughly as is then